### ISSUE TITLE
output Wikidata names with a newer metabolite ID mapping database

### ISSUE THREAD

**Issue by egonw** [BiGCAT-UM/DirectedSmallMoleculesNetwork#3]
Egon will make a new release today, fixing the missing name for ATP, and then upload the file for Jonathan to output a new TSV file.

**Comment by egonw** [comment]
New ID mapping database is ready (GDrive).

**Comment by egonw** [comment]
OK, now also released on Figshare: https://figshare.com/articles/Metabolite_BridgeDb_ID_Mapping_Database_20170621_/5135035

**Comment by JonathanMELIUS** [comment]
Here an example of the new output:

- Q80311	"SB 699551"@en
- **Q80863	"adenosine triphosphate"@en**
- Q82129	"Chlorophyll f"@en

Are the quotes and @en something intended ?

I found a special one:
Q1297015	"100122 Alpes Maritimes"@en

I have 152 560 ids/label in the output file. But there are 157 222 Wikidata ids in the new metabolites database. 
So there are 4 662 ids that doesn't have attributes. 
E.g:
- Q1013950
- Q1014238
- Q102165

**Comment by egonw** [comment]
No, the @en should have been removed :( I need to fix that... for now, it's ok... it's not a blocker

For those without labels, that happens a lot, as they may have a label in another language, but not English

**Comment by egonw** [comment]
(and Q1297015 needs some serious curation :)

**Comment by DeniseSl22** [comment]
WikiData ID's BridgeDb could not link to (from ChEBI ID) from dataset MTBLS265 (19 in total):

[CHEBI:15713](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:15713) : No Wikidata entry
[CHEBI:78697](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:78697)  :  Has Wikidata ID, compound name is D-fructose 6-phosphate in ChEBI, D-fructofuranose 6-phosphate in Wikidata
[CHEBI:14314](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:14314) : compound (famous D-glucose-6-phosphate) has two ChEBI's, this one was missing in Wikidata; I added it. 
[CHEBI:84055](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:84055) :  No wikidata entry
[CHEBI:17050](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:17050) : No wikidata entry
[CHEBI:24266](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:24266) : Has two ChEBIs, added missing one to WikiData.
[CHEBI:17234](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:17234) : Has two ChEBIs, added missing one to Wikidata.
[CHEBI:24898](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:24898) : isoleucine ( no stereoisomery defined); did not have a ChEBI ID yet, added this one.
[CHEBI:26986](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:26986) : threonine, did not have ChEBI ID yet, added this one.
[CHEBI:39033](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:39033) : Description from ChEBI: "Any member of a collection of zwitterionic buffer substances selected or devised for suitability in experimental biological systems according to a number of predetermined criteria." Added this ChEBi to [Q7119420](https://www.wikidata.org/wiki/Q7119420) 
[CHEBI:23818](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=) : Could be [Q27121426](https://www.wikidata.org/wiki/Q27121426); will check this with Egon before I add the ChEBI!
[CHEBI:29062](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:29062) : Has two ChEBIs, added missing one to Wikidata.
[CHEBI:5417](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:5417) : Has two ChEBIs, added missing one to Wikidata.
[CHEBI:15781](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:15781) : Does not have a Wikidata ID page.
[CHEBI:17311](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:17311) : Related to CHEBI:15781!!
[CHEBI:28683](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:28683) : Does not have a wikidata ID page.
[CHEBI:16436](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:16436) ; Uncharged state of CHEBI:49086, which has Q421308. Added this to Wikidata, however could not add qualifier :(.
[CHEBI:16870](https://www.ebi.ac.uk/chebi/searchId.do?chebiId=CHEBI:16870) : Does not have a wikidata ID page.

Removed one CHEBI which was double and did not have a WikiData ID.

**Comment by egonw** [comment]
For items that do not have a Wikidata item *and* have full stereochemistry defined, you can use the attached script. Just make a .smi file in the same folder and update the script to read that input file at the top of the script. Also, links [QuickStatement](https://tools.wmflabs.org/quickstatements/) to your Wikidata account (so that it has write access):

[createWDitemsFromSMILES.groovy.txt](https://github.com/BiGCAT-UM/DirectedSmallMoleculesNetwork/files/1138858/createWDitemsFromSMILES.groovy.txt)

**Comment by DeniseSl22** [comment]
Okay I will look at this! I removed the ChEBI IDs that do not have a Wikidata ID; since they will not occur in our network anyway and can make the calculation of the shortest path with neo4j difficult.
Removed IDs:
CHEBI:15713	[]
CHEBI:84055	[]
CHEBI:17050	[]
CHEBI:23818	[]
CHEBI:15781	[]
CHEBI:17311	[]
CHEBI:28683	[]
CHEBI:16870	[]

**Comment by DeniseSl22** [comment]
Hi Egon,

I checked all the additions etc. I made. There are two ChEBIs I could not strikethrough yet, I need your opinion on those. And one glucose ChEBI ID was behaving really bad:

"CHEBI:17234 : Has two ChEBIs, added missing one to Wikidata. Okay, this is a really bad one.... the previous ChEBI link refers to the same page (compare CHEBI:14313 with CHEBI:17234. Different identifier, goes to the same webpage. I think we need to let the @chebit know. I will remove CHEBI:14313 with the incorrect link in Wikidata!!"

**Comment by egonw** [comment]
Denise, what is your Wikidata account?

**Comment by egonw** [comment]
About CHEBI:14313 and CHEBI:17234... these are the same... the last is the primary ID, and the first is a secondary ID... I think this is how we should model it (and start a discussion about secundairy identifiers):

![image](https://user-images.githubusercontent.com/26721/28077265-92b1d1d4-6661-11e7-80d1-7e40ae72f03d.png)

**Comment by DeniseSl22** [comment]
My wikidata account name is the same as WPs: DeSl
